### PR TITLE
Sundry fixes/changes 

### PR DESCRIPTION
**Pull request by simo5**
#### Description

Found these things while investigating an issue, so putting them here or they will get lost

#### Checklist

- [x] Code modified for feature
- [x] Test suite updated with functionality tests
- [x] Test suite updated with negative tests
- [x] Documentation updated

#### Reviewer's checklist:

- [x] Any issues marked for closing are addressed
- [x] There is a test suite reasonably covering new functionality or modifications
- [x] This feature/change has adequate documentation added
- [x] Code conform to coding style that today cannot yet be enforced via the check style test
- [x] Commits have short titles and sensible commit messages
- [x] Coverity Scan has run if needed (code PR) and no new defects were found

### CODE REVIEW

File: src/tls.c [hunk 1]
```diff
@@ -107,14 +107,14 @@ struct {
...
-        TLS_PARAMS_ENTRY("secp256r1", "secp256r1", "EC", p256_group_id,
+        TLS_PARAMS_ENTRY("secp256r1", "prime256v1", "EC", p256_group_id,
                          p256_secbits, p256_mintls, p256_maxtls, p256_mindtls,
                          p256_maxdtls),
     },
     /* alias */
     {
         "P-256",
-        TLS_PARAMS_ENTRY("P-256", "secp256r1", "EC", p256_group_id,
+        TLS_PARAMS_ENTRY("P-256", "prim256v1", "EC", p256_group_id,
```
Comment (ueno) [inline src/tls.c:117]: "prim256v1" → "prime256v1"?

File: src/provider.c [hunk 2]
```diff
@@ -1388,6 +1398,37 @@ static struct p11prov_cfg_names {
     { "pkcs11-module-quirks" },
     { "pkcs11-module-cache-sessions" },
     { "pkcs11-module-encode-provider-uri-to-pem" },
+    { "pkcs11-module-block-operations" },
+};
+
+static struct p11prov_block_ops_names {
+    const char *name;
+    size_t op_num;
```
Comment (ueno) [inline src/provider.c:1406]: nit: Do you want to include op_num in the struct given it is monotonically increasing? If we can assume C99, it is also possible to use designator in the array initializer:
```c
const char *p11prov_block_ops_names[OSSL_OP__HIGHEST + 1] = {
    [OSSL_OP_DIGEST] = "digest",
    [OSSL_OP_CIPHER] = "cipher",
    ...
};
```

File: src/provider.c [hunk 3]
```diff
@@ -1176,10 +1177,35 @@ static const OSSL_ALGORITHM p11prov_decoders[] = {
     { NULL, NULL, NULL }
 };
 
+const char *p11prov_block_ops_names[OSSL_OP__HIGHEST + 1] = {
```
Comment (ueno) [inline src/provider.c:1180]: Add `static`?

File: docs/provider-pkcs11.7.md [hunk 4]
```diff
@@ -199,7 +199,24 @@ Default: none
 Example:
 
 ```pkcs11-module-quirks = no-deinit no-operation-state```
-(Disables deinitialization)
+(Disables deinitialization, blocks context duplication)
+
+## pkcs11-module-block-operations
+Allows to block specific "provider operations" even if the token actually
+supports the necessary mechanisms. This is useful to workaround cases
+where one wants to enforece use of the token for all operations by setting
```
Comment (ueno) [inline docs/provider-pkcs11.7.md:207]: "enforece" → "enforce"

File: docs/provider-pkcs11.7.md [hunk 5]
```diff
@@ -199,7 +199,24 @@ Default: none
...
+?provider=pkcs11 in the default properties but wants an exception for a
+specific type of operation like digests.
+NOTE: some opeations may depend on others or may be fundamental to the
```
Comment (ueno) [inline docs/provider-pkcs11.7.md:210]: "opeations" → "operations"

**Review by ueno — APPROVED** [review]
LGTM

File: src/provider.c (hunk 3 above)
Comment (simo5) [reply src/provider.c:1180]: Yeah, I guess I can declare it static, it won't be useful outside this file anyway.

**Comment by simo5** [comment]
@ueno thanks for the review!

**Pull request merged**